### PR TITLE
Revert "rootfs: handle nested procfs mounts for MS_MOVE" + add a test

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Reverts opencontainers/runc#2647 due to a security issue
Reopens https://github.com/opencontainers/runc/issues/2639

### CODE REVIEW

**Comment by cyphar** [comment]
This is actually a bug in `github.com/moby/sys/mountinfo` -- `info.Root` is always `""`. I've already submitted a fix (https://github.com/moby/sys/pull/50).

File: libcontainer/rootfs_linux.go [hunk 1]
```diff
@@ -809,46 +809,24 @@ func pivotRoot(rootfs string) error {
...
-	// filesystem is mounted) without any other locked mount points covering a
-	// subtree of the mount.
-	//
-	// So we try to unmount (or mount tmpfs on top of) any mountpoint which is
-	// a full mount of either sysfs or procfs (since those are the most
-	// concerning filesystems to us).
 	mountinfos, err := mountinfo.GetMounts(func(info *mountinfo.Info) (skip, stop bool) {
-		// Collect every sysfs and procfs filesystem, except for those which
-		// are non-full mounts or are inside the rootfs of the container.
-		if info.Root != "/" ||
```
Comment (cyphar) [inline libcontainer/rootfs_linux.go:832]: It's overkill to revert it -- a simpler patch is to just remove this `info.Root != "/"` clause which won't cause us to revert the fix (the `ENOENT` check later in the patch is sufficient to fix that bug).

Comment (AkihiroSuda) [reply libcontainer/rootfs_linux.go:832]: Let's revert it first and then revendor https://github.com/moby/sys/pull/50 when it is merged

**Comment by AkihiroSuda** [comment]
Added a new commit: `test: add "runc run --no-pivot must not expose bare /proc"`

File: libcontainer/rootfs_linux.go (hunk 1 above)
Comment (kolyshkin) [reply libcontainer/rootfs_linux.go:832]: I agree with @cyphar here -- for the reason of having a slightly less dramatic git history.

**Comment by cyphar** [comment]
#2655 removes just the `info.Root` check and cherry-picks 729e786.

**Pull request closed without merging**